shell/browser/api/atom_api_tray.cc

```diff
@@ -188,6 +188,20 @@ bool Tray::GetIgnoreDoubleClickEvents() {
 #endif
 }
 
+void Tray::SetVisible(bool visible) {
+#if defined(OS_MACOSX)
+  tray_icon_->SetVisible(visible);
+#endif
+}
+
+bool Tray::GetVisible() {
+#if defined(OS_MACOSX)
+  return tray_icon_->GetVisible();
+#else
+  return false;
```

Comment:
I would rather not expose the property on other platform than return `false`. Or at least return `true`, as you cannot hide the icon on other platforms.

Reply:
I agree this property should not be exposed to keep consistency with other platform-dependent APIs.

But the problem with property is, user has to explicitly test whether the property exists. Code like this would run without runtime exception:

```js
if (tray.visible) {
}
```

This is probably another reason not to do #12745.

Reply:
But `'visible' in tray` would work, right?

Reply:
Yeah there are multiple ways to test whether the property exists, but I think we should make the API hard-to-misuse, instead of hard-to-use-correctly.